wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/validator/DevfileIntegrityValidator.java

```diff
@@ -95,6 +96,9 @@ public void validateDevfile(Devfile devfile) throws DevfileFormatException {
           checkFieldNotSet(tool, "id", tool.getId());
           recipeTool = tool;
           break;
+        case DOCKERIMAGE_TOOL_TYPE:
+          // validation should be done via JSON Schema validator
```

Comment:
2 questions:
1) Can we have more than one dockerimage tool in devfile?  if no, where it is validated?
2) Can we have an K8S/OS tool and DI tool at the same time in devfile? if no, where it is validated?

Reply:
Nice catch. Will add the corresponding check

Reply:
The only one tool with type different from plugin no editor is supported. Added needed check and the corresponding unit test